core/src/Cabana_DeepCopy.hpp

```diff
@@ -126,6 +126,119 @@ inline void deep_copy(
     }
 }
 
+//---------------------------------------------------------------------------//
+/*!
+  \brief Create a mirror of the given AoSoA in the given memory
+  space. Same space specialization returns the input AoSoA.
+ */
```

Comment:
Kokkos provides both `create_mirror` and `create_mirror_view` functions.  The difference between the two is that `create_mirror` always allocate a new view while `create_mirror_view ` only creates a new view if the original one has a different memory space.  I am not a big fan of this naming convention.  `_view` does make sense in a way because it does give you access from the given execution space into the underlying multidimensional array.  But `View` is also what Kokkos call its md array but and the name does not suggest is that it retains shared ownership 
over that array...  Anyway here you replaced `_view` with `_aosoa` but I think it is wrong, you should still call it `_view` because that is what the semantics of your functions are.

Reply:
per discussion with @dalg24 I have reduced the number of functions implemented to cover what is needed for #88 - we can add more in the future as needed. Also we discussed the semantics of the word "Mirror" and "Mirror View" in Kokkos and how it applies here. The name I have chosen for our function represents those semantics but may be subject to change - hence inclusion in the Experimental namespace.